lib/mpl_toolkits/tests/test_mplot3d.py

```diff
@@ -420,6 +420,32 @@ def test_surface3d_shaded():
     ax.set_zlim(-1.01, 1.01)
 
 
+@mpl3d_image_comparison(['surface3d_masked.png'])
+def test_surface3d_masked():
+    fig = plt.figure()
+    ax = fig.add_subplot(projection='3d')
+    X = np.arange(1, 10, 1)
+    Y = np.arange(1, 10, 1)
+    X, Y = np.meshgrid(X, Y)
+    Z = X**3 + Y**3 - 500
+    Z = np.ma.masked_array(Z, Z < 0)
+    ax.plot_surface(X, Y, Z, lw=0, antialiased=False)
+    ax.set_zlim(-1, 1000)
+
+
+@mpl3d_image_comparison(['surface3d_masked_strides.png'])
+def test_surface3d_masked_strides():
+    fig = plt.figure()
+    ax = fig.add_subplot(projection='3d')
+    X = np.linspace(-1, 1)
+    Y = np.linspace(-1, 1)
+    X, Y = np.meshgrid(X, Y)
+    Z = X**2 + Y**2
+    Z[10:20, 10:20] = 0.0
```

Comment:
The location of the hole is a bit difficult to see; can you move it somewhere move visible?

Reply:
Thanks for the suggestion. I've updated the image with the hole in a more prominent place. I amended the commit though I have no preference on whether this is squash merged or not.